src/scripts/main.js

```diff
@@ -1,11 +1,27 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  const promise = new Promise((resolve) => {
+    element.addEventListener(eventName, () => {
+      resolve(
+        `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`,
+      );
+    });
+  });
+
+  promise.then((message) => {
+    return message;
+  });
```

Comment:
The `then` method here is unnecessary because it doesn't alter the promise chain or handle the resolved value in any meaningful way. You can remove this `then` block to simplify the code.